Engine/source/app/net/tcpObject.cpp

```diff
@@ -400,6 +405,29 @@ void TCPObject::send(const U8 *buffer, U32 len)
    Net::sendtoSocket(mTag, buffer, S32(len));
 }
 
+bool TCPObject::sendFile(const char* fileName)
+{
+	//Open the file for reading
+	FileStream readFile;
+	if(!readFile.open(fileName, Torque::FS::File::Read))
+	{
+		return false;
+	}
+
+	//Read each byte into our buffer
+	Vector<U8> buffer(readFile.getStreamSize());
+	U8 byte;
+	while(readFile.read(&byte))
+	{
+		buffer.push_back(byte);
+	}
+
+	//Send the buffer
+	send(buffer.address(), buffer.size());
```

Comment:
Interesting about the file size. I guess we never hit that limit, but make sense. I wrote a bunch of script code that basically used TCP object to do a bunch of HTTP stuff including opening up a SocketStream to send stuff to and from Google Earth. Most of the files we were sending were pretty small art files or like a JS file so there wasn't an issue. I will look into it and try to improve it, but it works in those limited cases.

And I know there is an HTTP object, but we needed to make rapid changes and the HTTP object didn't seem to:
1. do everything we need
2. had some issues with it working at all

As for the performance issue with reading the file, I'll improve that with your suggestion, thanks!